tools/generate_libdocs.py

```
@@ -0,0 +1,102 @@
+        output_file = output_dir / f"{library_name.lower()}.json"
+        
+        print(f"Generating libdoc for {library_name}...")
+        
+        # Generate libdoc JSON
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.json', delete=False) as temp_file:
+            temp_path = temp_file.name
+        
+        libdoc(library_name, temp_path, format='JSON')
+        
+        # Read and rewrite to output directory
+        with open(temp_path, 'r', encoding='utf-8') as f:
```

Ruff (0.15.5)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)